Exercise6/Logistics.java

```diff
@@ -0,0 +1,177 @@
+package Exercise6;
+
+import java.time.DayOfWeek;
+import java.time.Duration;
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Scanner;
+
+public class Logistics {
+
+	public static void main(String[] args) {
+		Scanner sc=new Scanner(System.in);
+		KamatchiTransport travel=new KamatchiTransport();
+		
+		Lorry lorry=new Lorry("Ashok Leyland","TN 56 T 0007",80);
+		
+		Driver driver=new Driver(34,"Ram","DL202000");
+		
+		System.out.println("Enter the year..");
+		int year=sc.nextInt();
+		
+		System.out.println("Enter the month..");
+		int month=sc.nextInt();
+		
+		System.out.println("Enter the day..");
+		int day=sc.nextInt();
+		
+		System.out.println("Enter the hours..");
+		int hrs=sc.nextInt();
+		
+		System.out.println("Enter the minutes..");
+		int min=sc.nextInt();
+		
+		Time time=new Time(year,month,day,hrs,min);
+		LocalDateTime datetime = time.getTime();
+		time.addLeaves();
+		
+		System.out.println("Enter the Kms to travel..");
+		int kms=sc.nextInt();
+		
+		System.out.println("The Lorry Will reach at "+travel.delivery(kms, datetime, lorry, driver));
+	}
+
+}
+abstract class Transport{
+	abstract public String delivery(int distance,LocalDateTime time,Lorry lorry,Driver driver);
+}
+class KamatchiTransport extends Transport{
+	@Override
+	public String delivery(int distance,LocalDateTime  time,Lorry lorry,Driver driver)
+	{
+		
+		int timetaken=(distance/lorry.max_speed)*60;
+		
+		while(Time.isLeave(time))
+			{
+				time=time.plusDays(1);
+				System.out.println("helloo");
+			}
+		
+			int remaining=Time.timeRemaining(time);
+			if(timetaken<=remaining)
+			{
+				time=time.plusMinutes(timetaken);
+				return time.format(DateTimeFormatter.ofPattern("yyyy-MM-dd HH:mm"))+"";
+			}
+			else
+			{
+				timetaken-=remaining;
+				time=time.plusDays(1);
+				while (timetaken >= 480) {
```

Comment:
Can you please add comments in places like this to explain the process and working?